### ISSUE TITLE
Incompatible TypedDicts resolution order not respected - Conformance with other type checkers

### ISSUE THREAD

**Issue by Daraan** [microsoft/pyright#11279]
**Describe the bug**
Given an incompatible subclassing of two TypedDicts, e.g. TD1 `a: int` and TD2 `a: str`. When subclassing we are used to that the first parent overwrites the children values, i.e. `class TD3(TD2, TD2)` should have `a: str` as annotation hint (even if incompatible).

[mypy](https://mypy-play.net/?gist=8ac04941114aaf154db1c2d17a2fb988), pyre and [ty](https://play.ty.dev/ea63ec6d-702e-4f11-8ca1-301785d94eaf) report key `a` as `str`, pyright as `int`. Reversing the bases reverses the results for all type checkers.

**Code or Screenshots**
Code sample in [pyright playground](https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAqiApgCYAiSAxjALABQDVANgIYDO7h5AjABREEZSjQCUALgZRpUVuMwo69GVABG89jBAMmbTtwBMAkhWowJUmXKibtymVXmolujlwLkAzAPIGANNw8ElAAdGE69KTEwFDAYGB81rYB6jZaAY7OwWEhEQxRMTCkfKJQALQAfNye8jl59CDEAG7ErMwA%2BvBCfEUlANoARKwDALqlAMRQxAAeQjRkaSABaGD4zhFxCQBUW72iogxAA)

```python
from typing import TypedDict

class TD1(TypedDict):
    a: int
    b: str

class TD2(TypedDict):
    a: str
    c: int

class TD3(TD2, TD1): ...  # incompatible subclass; ignore

def foo(a: str, b: str, c:int): ...


def td() -> TD3: ...


reveal_type(td()["a"]) # expected str, got int

foo(**td())  # likewise, error expected OK
```

**VS Code extension or command-line**
1.1.408

**Comment by erictraut** [comment]
Pyright is working correctly here. It honors the runtime behavior. The other type checkers do not.

```python
print(TD3.__annotations__)
{'a': <class 'int'>, 'c': <class 'int'>, 'b': <class 'str'>}
```

Pyright previously worked like the other type checkers here, but someone filed a bug pointing out that it didn't correctly follow the runtime, so I changed it.
